manager-bundle/src/ContaoManager/Plugin.php

```diff
@@ -407,6 +391,41 @@ private function addDefaultDoctrineMapping(array $extensionConfigs, ContainerBui
         return $extensionConfigs;
     }
 
+    /**
+     * Enables the SQL strict mode for PDO and MySQL drivers.
+     *
+     * @return array<string,array<string,array<string,array<string,mixed>>>>
+     */
+    private function enableStrictMode(array $extensionConfigs, ContainerBuilder $container): array
+    {
+        [$driver, $options] = $this->parseDbalDriverAndOptions($extensionConfigs, $container);
+
+        // Skip if driver is not supported
+        if (null === ($key = ['mysql' => 1002, 'mysqli' => 3][$driver] ?? null)) {
+            return $extensionConfigs;
+        }
+
+        // Skip if init command is already configured
+        if (isset($options[$key])) {
+            return $extensionConfigs;
+        }
+
+        // Enable strict mode
+        $extensionConfigs[] = [
+            'dbal' => [
+                'connections' => [
+                    'default' => [
+                        'options' => [
+                            $key => "SET SESSION sql_mode=(SELECT CONCAT(@@sql_mode, ',TRADITIONAL'))",
```

Comment:
```suggestion
                            $key => "SET SESSION sql_mode=IF(FIND_IN_SET('STRICT_TRANS_TABLES', @@sql_mode) + FIND_IN_SET('STRICT_ALL_TABLES', @@sql_mode), @@sql_mode, CONCAT(@@sql_mode, ',TRADITIONAL'))",
```

How about something like this? _(untested)_

Reply:
The alternative would be to just add `STRICT_TRANS_TABLES` only (as this is the minimum requirement).
But I would prefer the one from above.

```suggestion
                            $key => "SET SESSION sql_mode=(SELECT CONCAT(@@sql_mode, ',STRICT_TRANS_TABLES'))",
```

Reply:
Maybe we can reuse the existing check in the install tool? 

https://github.com/contao/contao/blob/c823a49d627cc40712b339fba198ded55a081f2a/installation-bundle/src/InstallTool.php#L277-L284

Reply:
Not sure if it is a good idea to send a `SET SESSION` query with two `FIND_IN_SET`s upon every request? I actually don‘t want Contao to send any `SET SESSION` query at all if my server is configured correctly.

Reply:
I would go with @ausi's proposal but switch `+` with `OR` to allow short circuiting. 

Init commands are not performance critical. They are bound to the connection, only evaluated once and - in this case - very cheap.

Reply:
Still I don‘t want Contao to automatically send a `SET SESSION` query. My server is configured correctly, please do not mess around with it.

Reply:
To be honest, I don't know why you're so opposed to this. We discussed and decided that Contao should enforce strict mode, we tested, that setting the session in init commands is no problem on several shared servers and I already invested way too much time in this PR and its predecessor. This is frustrating.

Reply:
```suggestion
                            $key => "SET SESSION sql_mode=CONCAT(@@sql_mode, IF(INSTR(@@sql_mode, 'STRICT_'), '', ',TRADITIONAL'))",
```

This would be shorter and only use a single `INSTR()` check. Takes about 6 microseconds on my system.

Reply:
> Takes about 6 microseconds

That was directly measured on the DB btw.

When creating a real PDO connection, I was unable to measure any time difference (probably because PDO sends an init command like `SET NAMES utf8` anyways...)